ios/RNSScreen.m

```diff
@@ -295,6 +297,63 @@ - (instancetype)initWithView:(UIView *)view
   return self;
 }
 
+- (UIStatusBarStyle)preferredStatusBarStyle
+{
+  UIViewController *child = [[self childViewControllers] lastObject];
+  if ([child isKindOfClass:[RNScreensNavigationController class]] || [child isKindOfClass:[RNScreensViewController class]]) {
+    return child.preferredStatusBarStyle;
+  }
+  RNSScreenStackHeaderConfig *config = [self findConfigForScreen];
+  return config && config.statusBarStyle ? [self statusBarStyleForRNSStatusBarStyle:config.statusBarStyle] : [self statusBarStyleForRNSStatusBarStyle:RNSStatusBarStyleAuto];
+}
+
+- (UIStatusBarAnimation)preferredStatusBarUpdateAnimation
+{
+  UIViewController *child = [[self childViewControllers] lastObject];
+  if ([child isKindOfClass:[RNScreensNavigationController class]] || [child isKindOfClass:[RNScreensViewController class]]) {
+    return child.preferredStatusBarUpdateAnimation;
+  }
+  RNSScreenStackHeaderConfig *config = [self findConfigForScreen];
+  return config && config.statusBarAnimation ? config.statusBarAnimation : UIStatusBarAnimationFade;
+}
+
+- (BOOL)prefersStatusBarHidden
+{
+  UIViewController *child = [[self childViewControllers] lastObject];
+  if ([child isKindOfClass:[RNScreensNavigationController class]] || [child isKindOfClass:[RNScreensViewController class]]) {
+    if ([child childViewControllerForStatusBarHidden]) {
+      return [child childViewControllerForStatusBarHidden].prefersStatusBarHidden;
+    }
+  }
+  RNSScreenStackHeaderConfig *config = [self findConfigForScreen];
+  return config && config.statusBarHidden ? config.statusBarHidden : NO;
+}
+
+- (RNSScreenStackHeaderConfig *)findConfigForScreen
+{
+  // if there is no child navigator and the parent is `RNSScreenContainer`, we should fallback to the parent's (that is not `RNSScreenContainer`) option
+  UIViewController *parent = [self parentViewController];
+  while ([parent isKindOfClass:[RNScreensViewController class]] || [parent isKindOfClass:[RNSScreen class]]) {
```

Comment:
Would it enter the loop if the `parent` was nil at the beginning? I think this while condition is not that bad since we iterate as long as the parent is one of those two kinds of VC. If we spotted something else, or we got to the top of the hierarchy, loop would stop.